src/network/networkd-route.c

```diff
@@ -773,13 +773,13 @@ static void log_route_debug(const Route *route, const char *str, const Link *lin
                 MultipathRoute *m;
 
                 ORDERED_SET_FOREACH(m, route->multipath_routes) {
-                        _cleanup_free_ char *buf = NULL, *joined = NULL;
+                        _cleanup_free_ char *buf = NULL;
                         union in_addr_union a = m->gateway.address;
 
                         (void) in_addr_to_string(m->gateway.family, &a, &buf);
-                        joined = strjoin(gw_alloc, gw_alloc ? "," : "", strna(buf), m->ifname ? "@" : "", strempty(m->ifname));
-                        if (joined)
-                                free_and_replace(gw_alloc, joined);
+                        (void) strextendf(&gw_alloc, "%s%s%s%s(weight:%u)",
+                                          gw_alloc ? "," : "", strna(buf), m->ifname ? "@" : "",
+                                          strempty(m->ifname), m->weight+1);
```

Comment:
Won't it be confusing if this is logged as "weight", but internally the same value - 1 is stored? Could we store the weight as the real value?

Reply:
Maybe add a comment explaining how the weight is stored and used so it's easier to grok for a casual user…